samples/benchmarks/coremark/src/main.c

```diff
@@ -170,8 +172,20 @@ static int button_init(void)
 	return ret;
 }
 
+static void main_thread_priority_cooperative_set(void)
+{
+	BUILD_ASSERT(CONFIG_MAIN_THREAD_PRIORITY >= 0);
+	k_thread_priority_set(k_current_get(), COOP_PRIO);
+}
+
 int main(void)
 {
+	/* Drivers need to be run from a non-blocking thread.
+	 * We need preemptive priority during init.
+	 * Later we prefer cooperative priority to ensure no interference with the benchmark.
```

Comment:
Maybe we could refer to the driver that actually requires preemptive priority for initialization. Unless I missed something it's not a generic requirement.

Reply:
We currently assume that for all SoCs we need to initialize **all** drivers from non-blocking context - so no